pallets/loans/src/tests/repay_loan.rs

```diff
@@ -812,3 +812,63 @@ fn with_unregister_price_id_and_oracle_not_required() {
 		);
 	});
 }
+
+#[test]
+fn with_external_pricing_and_overdue() {
+	new_test_ext().execute_with(|| {
+		let loan_id = util::create_loan(LoanInfo {
+			..util::base_external_loan()
+		});
+
+		let amount = ExternalAmount::new(QUANTITY, PRICE_VALUE);
+		util::borrow_loan(loan_id, PrincipalInput::External(amount));
+
+		let amount = ExternalAmount::new(Quantity::one(), PRICE_VALUE);
+		config_mocks(amount.balance().unwrap());
+
+		let repay_amount = RepaidInput {
+			principal: PrincipalInput::External(amount),
+			interest: 0,
+			unscheduled: 0,
+		};
+
+		let current_price = || {
+			ActiveLoanInfo::try_from((POOL_A, util::get_loan(loan_id)))
+				.unwrap()
+				.current_price
+				.unwrap()
+		};
+
+		// BEFORE: the loan not yet overdue
+		advance_time(YEAR - DAY);
+		assert_ok!(Loans::repay(
+			RuntimeOrigin::signed(BORROWER),
+			POOL_A,
+			loan_id,
+			repay_amount.clone()
+		));
+		assert!(current_price() < NOTIONAL);
+
+		// EXACT: the loan is just at matuyrity date
+		advance_time(DAY);
+
+		assert_ok!(Loans::repay(
+			RuntimeOrigin::signed(BORROWER),
+			POOL_A,
+			loan_id,
+			repay_amount.clone()
+		));
+		assert_eq!(current_price(), NOTIONAL);
+
+		// AFTER: the loan overpassing maturity date
+		advance_time(DAY);
+
+		assert_ok!(Loans::repay(
+			RuntimeOrigin::signed(BORROWER),
+			POOL_A,
+			loan_id,
+			repay_amount.clone()
+		));
+		assert_eq!(current_price(), NOTIONAL);
```

Comment:
Checked 3 cases, just before, just at the maturity date and just after

@mustermeiszer